fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/ProtectionsView.kt

```diff
@@ -63,11 +67,24 @@ class ProtectionsView(
         binding.trackingProtectionDetails.setOnClickListener {
             interactor.onTrackingProtectionDetailsClicked()
         }
+        updateDividerVisibility()
+    }
+
+    private fun updateDividerVisibility() {
+        if (!binding.trackingProtectionSwitch.isVisible &&
+            !binding.trackingProtectionDetails.isVisible &&
+            !binding.cookieBannerItem.isVisible
+        ) {
+            trackingProtectionDivider.visibility = View.GONE
+        } else {
+            trackingProtectionDivider.visibility = View.VISIBLE
```

Comment:
Let's use `trackingProtectionDivider.isVisible` to simplify this here.

Reply:
Done